src/coreclr/vm/threadsuspend.cpp

```diff
@@ -5733,7 +5733,9 @@ BOOL CheckActivationSafePoint(SIZE_T ip)
 
     // The criteria for safe activation is to be running managed code.
     // Also we are not interested in handling interruption if we are already in preemptive mode.
-    BOOL isActivationSafePoint = pThread != NULL &&
+    // Also we are not interested in handling interruption if we are single stepping
+    BOOL isActivationSafePoint = pThread != NULL && 
+        (pThread->m_StateNC & Thread::TSNC_DebuggerIsStepping) == 0 &&
```

Comment:
How come you used `pThread->m_stateNC` here, but in the change below, you just used `m_StateNC`?

Reply:
Because the other change is inside the class Thread in the method InjectActivation.